storage/src/tests/distributor/nodemaintenancestatstrackertest.cpp

```diff
@@ -64,38 +72,86 @@ NodeMaintenanceStatsTrackerTest::requestingNonExistingNodeGivesEmptyStats()
 {
     NodeMaintenanceStatsTracker tracker;
     NodeMaintenanceStats wanted;
-    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(0));
+    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(0, makeBucketSpace()));
 }
 
 void
 NodeMaintenanceStatsTrackerTest::statsAreTrackedPerNode()
 {
     NodeMaintenanceStatsTracker tracker;
     NodeMaintenanceStats wanted;
+    BucketSpace space(1);
 
-    tracker.incMovingOut(0);
+    tracker.incMovingOut(0, space);
     wanted.movingOut = 1;
-    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(0));
+    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(0, space));
     wanted.movingOut = 0;
-    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(1));
+    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(1, space));
 
-    tracker.incMovingOut(0);
+    tracker.incMovingOut(0, space);
     wanted.movingOut = 2;
-    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(0));
+    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(0, space));
 
-    tracker.incMovingOut(1);
+    tracker.incMovingOut(1, space);
     wanted.movingOut = 1;
-    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(1));
+    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(1, space));
 
-    tracker.incSyncing(1);
-    tracker.incCopyingIn(1);
-    tracker.incCopyingOut(1);
+    tracker.incSyncing(1, space);
+    tracker.incCopyingIn(1, space);
+    tracker.incCopyingOut(1, space);
     wanted.syncing = 1;
     wanted.copyingIn = 1;
     wanted.copyingOut = 1;
-    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(1));
+    CPPUNIT_ASSERT_EQUAL(wanted, tracker.forNode(1, space));
 }
 
-} // distributor
-} // storage
+void
+NodeMaintenanceStatsTrackerTest::statsAreTrackedPerBucketSpace()
+{
+    NodeMaintenanceStatsTracker tracker;
+    BucketSpace fooSpace(3);
+    BucketSpace barSpace(5);
+
+    tracker.incMovingOut(0, fooSpace);
+    assertBucketStats(1, 0, 0, 0, fooSpace, tracker);
+    assertBucketStats(barSpace, tracker);
+
+    tracker.incMovingOut(0, barSpace);
+    assertBucketStats(1, 0, 0, 0, fooSpace, tracker);
+    assertBucketStats(1, 0, 0, 0, barSpace, tracker);
+
+    tracker.incSyncing(0, fooSpace);
+    assertBucketStats(1, 1, 0, 0, fooSpace, tracker);
+    assertBucketStats(1, 0, 0, 0, barSpace, tracker);
+
+    tracker.incCopyingIn(0, fooSpace);
+    assertBucketStats(1, 1, 1, 0, fooSpace, tracker);
+    assertBucketStats(1, 0, 0, 0, barSpace, tracker);
+
+    tracker.incCopyingOut(0, fooSpace);
+    assertBucketStats(1, 1, 1, 1, fooSpace, tracker);
+    assertBucketStats(1, 0, 0, 0, barSpace, tracker);
+}
+
+void
+NodeMaintenanceStatsTrackerTest::assertBucketStats(BucketSpace bucketSpace,
```

Comment:
Could call this `assertEmptyBucketStats` or `assertZeroBucketStats` to clarify what it asserts